mcpscanner/core/static_analysis/native_analyzer.py

```
@@ -3017,6 +3061,367 @@ def _ts_parse_registration_args(
+
+        visit(root)
+        return names
+
+    def _ts_endpoint_names_in_array(self, array_node: "Node") -> List[str]:
+        """Return string tool names from object elements in an array literal."""
+        out: List[str] = []
+        for child in array_node.children:
+            if child.type not in ("object", "object_expression"):
+                continue
+            tool_name: Optional[str] = None
+            for field in self._TS_ENDPOINT_NAME_FIELDS:
```

Ruff (0.15.21)

[error] Import `field` from line 46 shadowed by loop variable

(F402)